dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import {
+  InputGroup,
+  InputGroupText,
+  DatePicker,
+  isValidDate,
+  Button,
+} from "@patternfly/react-core";
+import "./index.css";
+import { filterData } from "utils/filterDataset";
+import { dateFromUTCString } from "utils/constructDate";
+import { bumpToDate } from "utils/bumpDate";
+import { getTodayMidnightUTCDate } from "utils/getMidnightUTCDate";
+
+function DatePickerWidget({
+  dataArray,
+  setPublicData,
+  controllerName,
+  setDateRange,
+}) {
+  const [fromDate, setFromDate] = useState({});
+  const [toDate, setToDate] = useState(
+    bumpToDate(getTodayMidnightUTCDate(), 1)
+  );
+  const [strDate, setStrDate] = useState(
+    new Date().toLocaleDateString("fr-CA") //Return a YYYY-MM-DD string
+  );
+  const toValidator = (date) =>
+    date >= fromDate
+      ? ""
+      : "To date must be greater than or equal to from date";
+
+  const onFromChange = (_str, date) => {
+    const selectedDate = dateFromUTCString(_str);
+    setFromDate(isValidDate(date) ? selectedDate : {});
+    if (isValidDate(date)) {
+      if (date > new Date(strDate)) {
+        setToDate(bumpToDate(dateFromUTCString(_str), 1));
+        setStrDate(_str);
+      }
+    }
+  };
+
+  const filterByDate = () => {
+    setPublicData(filterData(dataArray, fromDate, toDate, controllerName));
+    setDateRange(fromDate, toDate);
+  };
+  return (
+    <InputGroup className="filterInputGroup">
+      <InputGroupText>Filter By Date</InputGroupText>
+      <DatePicker
+        onChange={onFromChange}
+        aria-label="Start date"
+        placeholder="YYYY-MM-DD"
+      />
+      <InputGroupText>to</InputGroupText>
+      <DatePicker
+        value={strDate}
+        onChange={(_str, date) => {
```

Comment:
So we're confident that the date picker will never call the `onChange` handler with a string that has a time zone value (which would break `dateFromUTCString`)? I don't see any clear statement about the format in the documentation; although their little sample widgets clearly reject anything but a basic date. I'm not entirely confident that means we're safe: are you? 😀

Reply:
Thanks. There ought to be a better way; but since that doesn't seem to be obvious at this point, we'll go with what we have.

In the future, it would be good to record assumptions like this with a comment in the code. We've mentioned before that starting a convention of writing JSDoc function comments would be good to help readers understand the parameters and behavior; it's also a good place to capture assumptions about inputs.